### PR TITLE
Updated Order Detail view to enable refund creation

### PR DESCRIPTION
**Pull request by clintonb**
Users can now select line items to refund from the dashboard's order detail view.

XCOM-357

@rlucioni @jimabramson

### CODE REVIEW

File: ecommerce/extensions/dashboard/orders/views.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.utils.safestring import mark_safe
+from django.utils.translation import ugettext_lazy as _
+from oscar.apps.dashboard.orders.views import OrderDetailView as CoreOrderDetailView
```
Comment (jimabramson) [inline ecommerce/extensions/dashboard/orders/views.py:5]: trivial: `BaseOrderDetailView` ?

File: ecommerce/extensions/dashboard/orders/views.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        refund = Refund.create_with_lines(order, lines)
+
+        if refund:
+            data = {
+                'link_start': '<a href="{}" target="_blank">'.format(
+                    reverse('dashboard:refunds:detail', kwargs={'pk': refund.pk})),
+                'link_end': '</a>',
+                'refund_id': refund.pk
+            }
+            message = _('%(link_start)sRefund #%(refund_id)d%(link_end)s created!') % data
```
Comment (jimabramson) [inline ecommerce/extensions/dashboard/orders/views.py]: That I could click on the confirmation message and go to the page for the new refund was not at all obvious (though now that I know, I know).  Did you consider 302-ing directly to that page upon success?  That would also get rid of the need to cobble together so much markup here.

Comment (jimabramson) [inline ecommerce/extensions/dashboard/orders/views.py]: alternatively, spell it out in the message, e.g. "Refund created!  Click here to view or process the refund."

File: ecommerce/extensions/dashboard/orders/tests.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Refund = get_model('refund', 'Refund')
+
+
+class OrderDetailViewTests(UserMixin, RefundTestMixin, TestCase):
+    def setUp(self):
+        super(OrderDetailViewTests, self).setUp()
+
+        # Staff permissions are required for this view
+        self.user = self.create_user(is_staff=True)
```
Comment (jimabramson) [inline ecommerce/extensions/dashboard/orders/tests.py:17]: is blocking access to this view for non-staff already covered by tests elsewhere?

File: ecommerce/extensions/dashboard/orders/views.py (hunk 2 above)
Comment (rlucioni) [inline ecommerce/extensions/dashboard/orders/views.py]: Minor: I would also encourage the use of `format()` here. As a translator, I'd be more likely to know that `{refund_id}` is a placeholder than `%(refund_id)d`.

**Comment by rlucioni** [comment]
Looks good. My minor comment aside, :+1: .

File: ecommerce/extensions/dashboard/orders/views.py (hunk 2 above)
Comment (clintonb) [inline ecommerce/extensions/dashboard/orders/views.py]: I've addressed both of your comments.

File: ecommerce/extensions/dashboard/orders/tests.py (hunk 3 above)
Comment (clintonb) [inline ecommerce/extensions/dashboard/orders/tests.py:17]: This should be covered by tests in Oscar.

**Comment by jimabramson** [comment]
:+1:

**Pull request merged**